### DISCUSSION TITLE
What is the proper way to use brew's clang(++)?

### DISCUSSION THREAD

**Discussion by eleanordoesntcode** [Homebrew/discussions#5746] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
/
```

### Output of `brew doctor`

```shell
/
```

### Description of issue

The question is in the title. I've got a formula that I'm currently rewriting for the new version of the software (ProjectM). I've realised that Apple's clang++ (which is on version 16) is too old for some features used, but by using Homebrew's clang++ (version 19.1.3), it worked. I'm now asking myself: what would be the proper way of explicitely requiring this compiler for the build, properly? As a sidenote: the build process uses cmake.

**Comment by p-linnane** [comment]
`depends_on "llvm" => :build`

**Reply by eleanordoesntcode** [reply]
Thanks! Wasn't sure whether this would use it in the build, since llvm is keg-only iirc.

**Reply by Bo98** [reply]
Adding `ENV.llvm_clang` to the start of your build will make it prioritise LLVM Clang, e.g. https://github.com/Homebrew/homebrew-core/blob/3f3347f2ed06e0716917c776e07defddde4fbef2/Formula/c/c-blosc2.rb#L31.
